sqlserver/datadog_checks/sqlserver/connection_errors.py

```diff
@@ -0,0 +1,148 @@
+# All rights reserved
+# Licensed under Simplified BSD License (see LICENSE)
+import re
+from enum import Enum
+
+try:
+    import pyodbc
+except ImportError:
+    pyodbc = None
+
+try:
+    import adodbapi
+    from adodbapi.apibase import OperationalError
+    from pywintypes import com_error
+except ImportError:
+    adodbapi = None
+
+
+class SQLConnectionError(Exception):
+    """Exception raised for SQL instance connection issues"""
+
+    pass
+
+
+class ConnectionErrWarning(Enum):
+    """
+    Denotes the various reasons a connection might fail.
+    """
+
+    unknown = "diagnosing-common-connection-issues"
+    login_failed_for_user = "sql-server-unable-to-connect-login-failed-for-user"
+    tcp_connection_failed = "sql-server-unable-to-connect-due-to-invalid-connection-string-attribute"
+    certificate_verify_failed = "ssl-provider-the-certificate-chain-was-issued-by-an-authority-that-is-not-trusted"
+    driver_not_installed = "picking-a-sql-server-driver"
+    dsn_not_found = "data-source-name-not-found-and-no-default-driver-specified"
+    ssl_security_error = 'sql-server-unable-to-connect-ssl-security-error-18'
```

Comment:
This enum's value seems more precise than its name (`ssl-security-error-18` vs `ssl_security_error`). I think it might make sense to have them match more closely (but I'm not sure which way is more correct)?

Reply:
:shrug: this is based on the name of this header here https://docs.datadoghq.com/database_monitoring/setup_sql_server/troubleshooting/#sql-server-unable-to-connect-ssl-security-error-18

Reply:
See my other comment.